Assets/NavMeshComponents/Scripts/NavMeshSurface.cs

```diff
@@ -341,10 +341,11 @@ List<NavMeshBuildSource> CollectSources()
             }
 
             if (m_IgnoreNavMeshAgent)
-                sources.RemoveAll((x) => (x.component != null && x.component.gameObject.GetComponent<NavMeshAgent>() != null));
+				sources.RemoveAll((x) => x.component != null && x.component is NavMeshAgent);
```

Comment:
This has been made in this way due to Its much more efficient a type check tan getting from the  "already obtained component" the component with he typed object, and checking if is not null

So
x.component != null && x.component is NavMeshAgent
Is better than
x.component != null && x.component.gameObject.GetComponent<NavMeshAgent>() != null
			
This even let you use this in an asynchronous way since no unity API is being called
I have this  oriented to async building, in my project, and it works fantastic

Is for that the second part of the condition is changed in both lines
So i see like a bad move not merging this, honestly